var/spack/repos/builtin/packages/exodusii/package.py

```diff
@@ -55,61 +94,85 @@ class Exodusii(CMakePackage):
 
-    depends_on("python@2.7:")
+    depends_on("python@3.0:")
+    conflicts("+shared", when="platform=windows")
+
+    def setup_run_environment(self, env):
+        env.prepend_path("PYTHONPATH", self.prefix.lib)
 
     def cmake_args(self):
         spec = self.spec
+        from_variant = self.define_from_variant
+        define = self.define
+
+        if self.spec.satisfies("@2022-10-14:"):
+            project_name_base = "Seacas"
+        else:
+            project_name_base = "SEACASProj"
+
+        options = []
 
-        cc_path = spec["mpi"].mpicc if "+mpi" in spec else self.compiler.cc
-        cxx_path = spec["mpi"].mpicxx if "+mpi" in spec else self.compiler.cxx
-
-        options = [
-            # General Flags #
-            "-DSEACASProj_ENABLE_SEACASExodus=ON",
-            "-DSEACASProj_ENABLE_TESTS=ON",
-            "-DBUILD_SHARED_LIBS:BOOL=ON",
-            "-DTPL_ENABLE_Netcdf:BOOL=ON",
-            "-DHDF5_NO_SYSTEM_PATHS=ON",
-            "-DSEACASProj_SKIP_FORTRANCINTERFACE_VERIFY_TEST:BOOL=ON",
-            "-DSEACASProj_ENABLE_CXX11:BOOL=OFF",
-            "-DSEACASProj_ENABLE_Zoltan:BOOL=OFF",
-            "-DNetCDF_DIR:PATH={0}".format(spec["netcdf-c"].prefix),
-            # MPI Flags #
-            "-DTPL_ENABLE_MPI={0}".format("ON" if "+mpi" in spec else "OFF"),
-            "-DCMAKE_C_COMPILER={0}".format(cc_path),
-            "-DCMAKE_CXX_COMPILER={0}".format(cxx_path),
-        ]
-        if "+fortran" in spec:
-            fc_path = spec["mpi"].mpifc if "+mpi" in spec else self.compiler.f90
+        # #################### Base Settings #######################
+        # Only want to enable the Exodus library.  If want anything else, use the seacas package.
+        options.extend(
+            [
+                define(project_name_base + "_ENABLE_ALL_PACKAGES", False),
+                define(project_name_base + "_ENABLE_ALL_OPTIONAL_PACKAGES", False),
+                define(project_name_base + "_ENABLE_SECONDARY_TESTED_CODE", False),
+                define(project_name_base + "_ENABLE_SEACASExodus", True),
+                from_variant(project_name_base + "_ENABLE_SEACASExodus_for", "fortran"),
+                from_variant(project_name_base + "_ENABLE_SEACASExoIIv2for32", "fortran"),
+                define(project_name_base + "_HIDE_DEPRECATED_CODE", False),
+                from_variant("CMAKE_INSTALL_RPATH_USE_LINK_PATH", "shared"),
+                from_variant("BUILD_SHARED_LIBS", "shared"),
+                from_variant("SEACASExodus_ENABLE_THREADSAFE", "thread_safe"),
+                from_variant("TPL_ENABLE_Pthread", "thread_safe"),
+                from_variant(project_name_base + "_ENABLE_Fortran", "fortran"),
+            ]
+        )
+        if "~shared" in self.spec and not is_windows:
+            options.append(self.define(project_name_base + "_EXTRA_LINK_FLAGS", "z;dl"))
+        options.append(from_variant("TPL_ENABLE_MPI", "mpi"))
+        if "+mpi" in spec and not is_windows:
```

Comment:
The check here should technically be for `msmpi` being the MPI provider, not just the Windows platform.
I realize you're just porting the logic that's already there (to be clear, this is my mistake), so if you'd like to get this PR in as is, I can be responsible for that change.

Reply:
If you don't mind, that would be good.  

We should also look at getting shared library support for exodus working.  There are `EXPORT` defines that decorate each function, but I don't think they are ever activated on a Windows build...  I don't have much access or knowledge on Windows, so any help would be appreciated (understand if that is not something you need...)

Reply:
We should also make sure that the `msmpi` change is also done in the seacas package to keep them consistent.

Reply:
> I don't have much access or knowledge on Windows, so any help would be appreciated (understand if that is not something you need...)

I'm always happy to help with Windows work! And exodusII is a pretty important package for a number of paraview related tasks, so I'm happy to help. Feel free to reach out on the Spack Slack to iterate on this (same user name).

> We should also make sure that the msmpi change is also done in the seacas package to keep them consistent.

Yes, I've got a local branch that's adding intel-mpi support to Windows, so everywhere the check is for mpi and windows will be refactored. The goal of the effort is actually to support Exodusii + Fortran + MPI on Windows, so Exodusii and Seacas are high on my support list.

 Since there is a good deal of shared CMake/general code between Exodusii, Seacas, and Trilinos, do you think it would make sense to condense the Spack configuration for common options to some centralized parent class?

Reply:
I'm fine doing that with Exodusii and Seacas, but I wouldn't want to include Trilinos just because it would make it more difficult to update if I had to get consensus/permission from Trilinos...  Prefer to do it only on the packages I have control over...

Reply:
Oh sorry, I saw you maintain Seacas and Exodus and just assumed. 
We can limit to Seacas and Exodus, that probably makes more sense anyway.